src/ApiClient.php

```diff
@@ -20,12 +21,14 @@ class ApiClient extends GraphQLClient
      * ApiClient constructor.
      * @param ClientInterface $httpClient
      * @param string $baseUrl
-     * @param string $token
+     * @param array $requestOptions
      */
-    public function __construct(ClientInterface $httpClient, string $baseUrl)
+    public function __construct(ClientInterface $httpClient, string $baseUrl, array $requestOptions = [])
     {
         $this->httpClient = $httpClient;
         $this->baseUrl = $baseUrl;
+        $this->connectionTimeout = Arr::get($requestOptions,'connect_timeout', 0);
```

Comment:
why not just `$requestOptions['connection_timeout']` ?

Reply:
Could be an away too, I am using helper method to have clean code.